lock-api/src/main/java/com/palantir/lock/client/TimelockRpcClient.java

```diff
@@ -0,0 +1,149 @@
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.WaitForLocksRequest;
+import com.palantir.lock.v2.WaitForLocksResponse;
+import com.palantir.logsafe.Safe;
+import com.palantir.timestamp.TimestampRange;
+
+/**
+ * Interface describing timelock endpoints to be used by feign client factories to create raw clients.
+ *
+ * If you are adding a new endpoint that is a modified version of an existing one, please use the existing naming scheme
+ * in which you add a new version number, rather than describing the additional functionality (i.e. refresh-locks-v2
+ * rather than leasable-refresh-locks)
+ */
+
+@Path("/timelock")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface TimelockRpcClient {
+
+    @POST
+    @Path("fresh-timestamp")
+    long getFreshTimestamp();
+
+    @POST
+    @Path("fresh-timestamps")
+    TimestampRange getFreshTimestamps(@Safe @QueryParam("number") int numTimestampsRequested);
+
+    @POST
+    @Path("lock-immutable-timestamp")
+        // TODO (jkong): Can this be deprecated? Are there users outside of Atlas transactions?
+    LockImmutableTimestampResponse lockImmutableTimestamp(IdentifiedTimeLockRequest request);
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#startAtlasDbTransactionV3(
+     * StartIdentifiedAtlasDbTransactionRequest)} instead; ignore the partition information if it is not useful for you.
+     */
+    @POST
+    @Path("start-atlasdb-transaction")
+    @Deprecated
+    StartAtlasDbTransactionResponse startAtlasDbTransaction(IdentifiedTimeLockRequest request);
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#startAtlasDbTransactionV3(
+     * StartIdentifiedAtlasDbTransactionRequest)} instead.
+     */
+    @POST
+    @Path("start-identified-atlasdb-transaction")
+    @Deprecated
+    StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction(
+            StartIdentifiedAtlasDbTransactionRequest request);
+
+    @POST
+    @Path("immutable-timestamp")
+    long getImmutableTimestamp();
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#lockV2(LockRequest) instead}.
+     */
+    @POST
+    @Path("lock")
+    @Deprecated
+    LockResponse lock(LockRequest request);
+
+    @POST
+    @Path("await-locks")
+    WaitForLocksResponse waitForLocks(WaitForLocksRequest request);
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#refreshLockLeasesV2(Set)} instead}.
+     */
+    @POST
+    @Path("refresh-locks")
+    @Deprecated
+    Set<LockToken> refreshLockLeases(Set<LockToken> tokens);
+
+    @POST
+    @Path("refresh-locks-v2")
+    LeasableRefreshLockResponse refreshLockLeasesV2(Set<LockToken> tokens);
+
+    @POST
+    @Path("lock-v2")
+    LeasableLockResponse lockV2(LockRequest request);
+
+    @POST
+    @Path("start-atlasdb-transaction-v3")
+    LeasableStartIdentifiedAtlasDbTransactionResponse startAtlasDbTransactionV3(
```

Comment:
is adding a new endpoint strictly necessary? can `LeasableStartIdentifiedAtlasDbTransactionResponse` extend `StartIdentifiedAtlasDbTransactionResponse`? If that were the case, then unknown properties will be ignored on old clients, and you can just change the type to include new fields.

Reply:
> unknown properties will be ignored on old clients

you sure about that? 🤣

Reply:
hoping/praying? 😅